test/utils/deployment.ts

```diff
@@ -0,0 +1,172 @@
+        });
+        up.stderr.on('data', chunk => {
+            neatlyPrintChunk('[docker-compose up | error]', chunk);
+        });
+    } else if (logType === LogType.File) {
+        const logStream = fs.createWriteStream(`${apiRootDir}/dependency_logs`, { flags: 'a' });
+        const errorStream = fs.createWriteStream(`${apiRootDir}/dependency_errors`, { flags: 'a' });
+        up.stdout.pipe(logStream);
+        up.stderr.pipe(errorStream);
+    }
+
+    // Wait for the dependencies to boot up.
+    await waitForDependencyStartupAsync(up);
+}
+
+/**
+ * Tears down 0x-api's dependencies.
+ */
+export async function teardownDependenciesAsync(): Promise<void> {
+    const down = spawn('docker-compose', ['down'], {
+        cwd: apiRootDir,
+    });
+    await new Promise<void>(resolve => {
+        down.on('close', () => {
+            resolve();
+        });
+    });
+    await rimrafAsync(`${apiRootDir}/0x_mesh`);
+    await rimrafAsync(`${apiRootDir}/postgres`);
+}
+
+function neatlyPrintChunk(prefix: string, chunk: Buffer): void {
+    const data = chunk.toString().split('\n');
+    data.filter((datum: string) => datum !== '').map((datum: string) => {
+        log.log(prefix, datum.trim());
+    });
+}
+
+async function waitForApiStartupAsync(logStream: ChildProcessWithoutNullStreams): Promise<void> {
+    return new Promise<void>(resolve => {
+        logStream.stdout.on('data', (chunk: Buffer) => {
+            const data = chunk.toString().split('\n');
+            for (const datum of data) {
+                if (/API (HTTP) listening on port 3000!/.test(datum)) {
+                    resolve();
+                }
+            }
+        });
+    });
+}
+
+async function waitForDependencyStartupAsync(logStream: ChildProcessWithoutNullStreams): Promise<void> {
+    return new Promise<void>(resolve => {
+        const hasSeenLog = [0, 0, 0];
+        logStream.stdout.on('data', (chunk: Buffer) => {
+            const data = chunk.toString().split('\n');
+            for (const datum of data) {
+                if (hasSeenLog[0] < 2 && /.*mesh.*started HTTP RPC server/.test(datum)) {
+                    hasSeenLog[0]++;
+                } else if (hasSeenLog[1] < 2 && /.*mesh.*started WS RPC server/.test(datum)) {
+                    hasSeenLog[1]++;
+                } else if (
+                    // NOTE(jalextowle): Because the `postgres` database is deleted before every
+                    // test run, we must skip over the "autovacuming" step that creates a new
+                    // postgres table.
+                    hasSeenLog[2] < 2 &&
+                    /.*postgres.*database system is ready to accept connections/.test(datum)
+                ) {
+                    hasSeenLog[2]++;
+                }
+
+                if (hasSeenLog[0] === 1 && hasSeenLog[1] === 1 && hasSeenLog[2] === 2) {
+                    resolve();
+                }
+            }
+        });
+    });
```

Comment:
I could see this never resolving if one of these dependencies is updated and the log changes. Should we reject with a helpful msg if this takes say, more than 20seconds? You'd know the best value for the seconds.